### PR TITLE
Updated: Dockerfile & README.md

### PR DESCRIPTION
**Pull request by GoliathLabs**
Good evening,
I cleaned up the Dockerfile as well as the README.md and added a few improvements.
I would love to hear from you. Feedback is always welcome as I have not that much experience writing Dockerfiles yet. :)

With best regards,
thegoliath

### CODE REVIEW

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-if [ ! -f "/root/.config/supertuxkart/config-0.10/stkservers.db" ]; then
-    touch /root/.config/supertuxkart/config-0.10/stkservers.db
```
Comment (juanelas) [inline docker-entrypoint.sh:3]: If you remove this file, you won't be able any more to use a db file, isn't it? Have you tested it? Sorry I don't have time now to do it my self (I may in a few weeks)

File: Dockerfile [hunk 2]
```diff
@@ -1,39 +1,41 @@
...
-            break; \
-        else \
-            echo "[$i] reconnecting to SVN"; \
-            svn cleanup stk-assets; \
-        fi; \
-    done && \
-    if [ "$i" -eq "100" ]; then \
-        echo "too many retries for SVN checkout" 1>&2; \
-        exit 1; \
-    fi && \
```
Comment (juanelas) [inline Dockerfile:21]: I know this code seems ugly but in my experience the assets svn servers often fails and this is sometimes required for a successful build. I rather keep it.

**Comment by juanelas** [comment]
Thank you very much for contributing!!!

File: Dockerfile [hunk 3]
```diff
@@ -1,39 +1,41 @@
...
+                       libenet-dev \
+                       libssl-dev \
+                       libsqlite3-dev \
+                       pkg-config \
+                       subversion \
+                       zlib1g-dev \
+                       ca-certificates
+
+# Using releases to make builds reproducible
+RUN git clone -b ${STK_VERSION} --depth=1 https://github.com/supertuxkart/stk-code.git
```
Comment (juanelas) [inline Dockerfile:22]: Great contribution here!

File: docker-entrypoint.sh [hunk 4]
```diff
@@ -1,5 +1,5 @@
...
-fi
-/usr/local/bin/supertuxkart $@
+if [ -n ${USERNAME} -a -n ${PASSWORD} ]; then
+    supertuxkart --init-user --login=${USERNAME} --password=${PASSWORD}
+fi &&
+/usr/local/bin/supertuxkart --server-config=server_config.xml $@
```
Comment (juanelas) [inline docker-entrypoint.sh:5]: There are some (probably unlikely) cases where you don't need the server_config.xml file. Maybe consider adding the file if it has been mounted as a volume?

File: docker-entrypoint.sh (hunk 1 above)
Comment (GoliathLabs) [reply docker-entrypoint.sh:3]: I will test things out this week :)

**Pull request closed without merging**